### DISCUSSION TITLE
Improving MCMC corner plot for DESIJ1554+0443

### DISCUSSION THREAD

**Discussion by ZareefJafar** [AstroBridge/BDLensing#201] [Q&A] [answered]
While creating a corner plot after MCMC for DESIJ1554+0443 ([notebook](https://github.com/ZareefJafar/BDLensing/blob/main/lens_systems/DESIJ1554%2B0443/lens_model_DESIJ1554%2B0443.ipynb)) I am getting a warning message  `WARNING:root:Too few points to create valid contours`. As discussed with @ajshajib it may be because of the sharp or rapid density changes in most of the 2-D projections as seen in the below corner plot (1st image). In the trace plot (2nd image) it seems all the parameters converged. Some lights are also leaking at the center of the source light (4th image). Please let me know if anyone has any suggestions. Thank you. 

Total sampling iterations: 600000 
N_walkers:        200
N_params:        22
N_step:             3000 for the trace plot
reduced X^2 :    0.87785

Parameter values after fit:
{'kwargs_lens': [{'theta_E': 1.5067581095825522, 'gamma': 1.8904822603698728, 'e1': -0.040309866222832426, 'e2': -0.10184439037838816, 'center_x': -0.29242504129128527, 'center_y': -0.11159130670709981}, {'gamma1': 0.10688744929291215, 'gamma2': 0.03494391561157853, 'ra_0': 0, 'dec_0': 0}], 
'kwargs_source': [{'amp': 1, 'R_sersic': 0.19919983612656295, 'n_sersic': 1.0, 'e1': -0.3879207176869724, 'e2': -0.35214145198407454, 'center_x': -0.4708827438889511, 'center_y': -0.09094553718817973}, {'amp': 1, 'n_max': 5, 'beta': 0.14540223843539649, 'center_x': -0.4708827438889511, 'center_y': -0.09094553718817973}], 
'kwargs_lens_light': [{'amp': 1, 'R_sersic': 0.49618063160543213, 'n_sersic': 4.0, 'e1': -0.06686542716305341, 'e2': -0.2328462184368, 'center_x': -0.3302467993257423, 'center_y': -0.09515750590790524}, {'amp': 1, 'R_sersic': 0.5284770963906795, 'n_sersic': 4.0, 'e1': -0.06069009745934361, 'e2': -0.2116975025728977, 'center_x': -0.3302467993257423, 'center_y': -0.09515750590790524}], 
'kwargs_ps': [], 'kwargs_special': {}, 'kwargs_extinction': []}

![corner](https://github.com/ajshajib/BDLensing/assets/49613484/2b2e96a5-65db-49db-bee8-632e0b9f4fd0)

![tracePlot](https://github.com/ajshajib/BDLensing/assets/49613484/d6db1cce-28f5-4448-b606-90a247fbc90f)

![reconstructed_images](https://github.com/ajshajib/BDLensing/assets/49613484/35957d79-ae89-43cf-be83-8b0b742c17be)

![source_light](https://github.com/ajshajib/BDLensing/assets/49613484/6dc380ff-e008-417c-8220-f603e56c0134)

**Comment by ZareefJafar** [comment]
@rafee-adnan @JobairHasan  @Rajit13 kindly let me know if you have any idea about this. Thank you.

**Reply by Rajit13** [reply]
It seems you already achieved convergence in `traceplot`. I encountered the same problem when I tried to make `corner` using only a few iterations. Take atleast 500 iterations to make a proper corner plot. According to your code you only took `burnin = 30`. Try `burnin=2500`. 

let me know the update.

**Reply by ZareefJafar** [reply]
Thank you @Rajit13 for the feedback. I already tested with higher `burnin` values (1500, 2000, 2500, 3000 etc ) for corner plots and it did not make a significant difference. I applied some burnin (200-3000) during the last 2-3 mcmc run even after converging. So, the higher value of burnin in the corner plot did not make any difference.  But I again applied `burnin=2500 ` as you said and here is the result. `WARNING:root:Too few points to create valid contours` remains no matter what the `burnin` value is. 

![corner](https://github.com/ajshajib/BDLensing/assets/49613484/af27e393-e2d9-4cf2-ab0b-767f62f78502)

**Comment by Rajit13** [comment]
@ZareefJafar what's the update regarding your problem?

**Reply by ZareefJafar** [reply, marked as answer]
Resolved this problem by adjusting burnin and thin
